demo/demo.js

```diff
@@ -92,12 +92,24 @@ $(function () {
     "clt": {"uri": "clt.nc", "variable": "clt"},
     "u": {"uri": "clt.nc", "variable": "u"},
     "v": {"uri": "clt.nc", "variable": "v"},
+    // execute a list of operations on the variable
+    // for subSlice, null (or undefined) means the current value for being or end index
+    "u_subset": {"uri": "clt.nc", "variable": "u",
+                 "operations": [{"subRegion": {'longitude1': [60, 180], 'latitude1': [0, 90]}},
+                                {"subSlice": {'longitude1': [null,null,2], 'latitude1': [null,null,2]}}]
+                },
+    "v_subset": {"uri": "clt.nc", "variable": "v",
+                 "operations":
+                 [{"subRegion": {'longitude2': [60, 180], 'latitude2': [0, 90]}},
+                  {"subSlice": {'longitude1': [,,2], 'latitude1': [,,2]}}]
```

Comment:
is [,,2] a correct syntax in js? Or should it be [null,null,2] as above?

Reply:
I am no javascript expert, but I think the valuel gets set  to 'undefined'. This is how I had it initially, and then I wanted to try it with null as well. Both versions end up None on the server so it seems fine.

Reply:
@danlipsa yes () reads the WHOLE variable in memory it could be huge. Then you subset further the variable on the next lines. I think we risk running into memeory errors here if a user is trying to access a small portion of very huge variables (e.g user wants one month of hourly data out of a file that contains decades of it). The whole variable would not fit in memory.

Reply:
Sounds good. I replaced () with []